packages/paste-core/components/heading/src/index.tsx

```diff
@@ -10,66 +11,90 @@ export type HeadingStyle =
   | 'headingStyle50'
   | 'headingStyle60';
 
-interface HeadingProps {
+interface Heading {
   as: asTags;
-  headingStyle?: HeadingStyle;
+  id?: never;
+  className?: never;
+  children: NonNullable<React.ReactNode>;
+  variant?: HeadingStyle;
 }
 
 function getHeadingStyles(headingStyle?: HeadingStyle): {} {
   switch (headingStyle) {
     case 'headingStyle60':
       return {
+        marginBottom: 'space70',
         fontSize: 'fontSize90',
-        fontWeight: 'fontWeightNormal',
-        lineHeight: 'lineHeight70',
-      };
-    case 'headingStyle50':
-      return {
-        fontSize: 'fontSize80',
-        fontWeight: 'fontWeightNormal',
-        lineHeight: 'lineHeight40',
+        fontWeight: 'fontWeightSemibold',
+        lineHeight: 'lineHeight60',
       };
     case 'headingStyle40':
       return {
-        fontSize: 'fontSize70',
-        fontWeight: 'fontWeightSemiBold',
-        lineHeight: 'lineHeight40',
+        marginBottom: 'space50',
+        fontSize: 'fontSize60',
+        fontWeight: 'fontWeightSemibold',
+        lineHeight: 'lineHeight70',
       };
     case 'headingStyle30':
       return {
-        fontSize: 'fontSize60',
-        fontWeight: 'fontWeightSemiBold',
-        lineHeight: 'lineHeight30',
+        marginBottom: 'space40',
+        fontSize: 'fontSize40',
+        fontWeight: 'fontWeightSemibold',
+        lineHeight: 'lineHeight50',
       };
     case 'headingStyle20':
       return {
-        fontSize: 'fontSize50',
-        fontWeight: 'fontWeightSemiBold',
-        lineHeight: 'lineHeight30',
+        marginBottom: 'space30',
+        fontSize: 'fontSize30',
+        fontWeight: 'fontWeightSemibold',
+        lineHeight: 'lineHeight20',
       };
     case 'headingStyle10':
       return {
-        fontSize: 'fontSize30',
-        fontWeight: 'fontWeightSemiBold',
-        lineHeight: 'lineHeight20',
+        marginBottom: 'space30',
+        fontSize: 'fontSize20',
+        fontWeight: 'fontWeightSemibold',
+        lineHeight: 'lineHeight10',
       };
+    /**
+     * headingStyle50 is out of order because its also default.
+     * Default is at the bottom of switch statement for readability.
+     */
+    case 'headingStyle50':
     default:
       return {
-        fontSize: 'fontSize80',
-        lineHeight: 'lineHeight40',
+        marginBottom: 'space60',
+        fontSize: 'fontSize70',
+        fontWeight: 'fontWeightSemibold',
+        lineHeight: 'lineHeight50',
       };
   }
 }
 
-const Heading: React.FC<HeadingProps> = ({as, headingStyle, children}) => (
-  <Text as={as} textColor="colorText" {...getHeadingStyles(headingStyle)}>
-    {children}
-  </Text>
-);
+const Heading: React.FC<Heading> = ({as, children, variant}) => {
+  return (
+    <Text as={as} textColor="colorText" {...getHeadingStyles(variant)}>
```

Comment:
We'll have to set `display="block"` here. If you set the as to a `label` or `span` the default display property from headings supplied by the browser will disappear and may cause undesired or unexpected styling issues for the consumer.